src/transformers/models/vits/modeling_vits.py

```diff
@@ -1435,7 +1435,7 @@ def forward(
         if self.config.num_speakers > 1 and speaker_id is not None:
             if not 0 <= speaker_id < self.config.num_speakers:
                 raise ValueError(f"Set `speaker_id` in the range 0-{self.config.num_speakers - 1}.")
-            speaker_embeddings = self.embed_speaker(torch.tensor([speaker_id])).unsqueeze(-1)
+            speaker_embeddings = self.embed_speaker(torch.tensor([speaker_id], device=self.device)).unsqueeze(-1)
```

Comment:
Let's only convert the speaker id to a torch tensor **if** it's an `int`. If the user already has passed a torch tensor, then we expect them to have passed it on the correct device.
```suggestion
            if isinstance(speaker_id, int):
                speaker_id = torch.tensor([speaker_id], device=self.embed_speaker.device)
            speaker_embeddings = self.embed_speaker(speaker_id).unsqueeze(-1)
```

Reply:
Sounds good to me!